### PR TITLE
Add vpc access connector resource and vpc_connector attribute to cloud functions

### PR DESCRIPTION
**Pull request by megan07**
VPC Access Connectors are now in beta.  Issue https://github.com/terraform-providers/terraform-provider-google/issues/3401 has requested the resource be available in the Terraform Google provider, and also available when creating a CloudFunctions Function.  I have added the `google_vpc_access_connector` resource and added the `vpc_connector` option on the `google_cloudfunctions_function` resource.

# Release Note for Downstream PRs (will be copied)
```releasenote
added the `google_vpc_access_connector` resource and the `vpc_connector` option on the `google_cloudfunctions_function` resource.
```

### CODE REVIEW

File: products/vpcaccess/api.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          - ERROR
+          - UPDATING
+      - !ruby/object:Api::Type::Integer
+        name: minThroughput
+        description: |
+          Minimum throughput of the connector in Mbps. Default and min is 200.
+      - !ruby/object:Api::Type::Integer
+        name: maxThroughput
+        description: |
+          Maximum throughput of the connector in Mbps. Default is 1000.
```
[the remark this answers is not in the document]
Comment (megan07) [reply products/vpcaccess/api.yaml]: I had this in there, but then I got an error that the minimum throughput must be lower than the maximum.  So I went to the console and it defaults to 1000.

File: third_party/terraform/resources/resource_cloudfunctions_function.go [hunk 2]
```diff
@@ -195,6 +195,11 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"vpc_connector": {
```
[the remark this answers is not in the document]
Comment (megan07) [reply third_party/terraform/resources/resource_cloudfunctions_function.go]: That's how I understand it?  Do I need to change something here for that?

[the remark this answers is not in the document]
Comment (rileykarson) [reply third_party/terraform/resources/resource_cloudfunctions_function.go]: Nope! I just wanted to confirm my understanding was correct.

[the remark this answers is not in the document]
Comment (JasperWoo) [reply third_party/terraform/resources/resource_cloudfunctions_function.go]: @rileykarson @megan07 I would love this to come alive, but something to add here. 

I only see `--vpc-connector` option in `gcloud  beta functions`. Will this affect? Can we test it before rolling out?

[the remark this answers is not in the document]
Comment (paddycarver) [reply third_party/terraform/resources/resource_cloudfunctions_function.go]: It appears the field is available in the v1 API for cloud functions: https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions

It's just the vpcaccess API that's in beta at the moment, I think? I think it's valid to have a beta resource referenced from a v1 resource.

**Pull request merged**